core/src/main/java/org/fao/geonet/kernel/setting/SettingManager.java

```diff
@@ -195,10 +199,27 @@ public String getValue(String path, boolean nullable) {
             Log.error(Geonet.SETTINGS, "  Requested setting with name: " + path + "  not found. Add it to the settings table.");
             return null;
         }
+
         String value = se.get().getValue();
+
+        // This case occurs during the application startup, before the encryptor is initialized:
+        // value is null and storedValue has the correct value in this case.
+        // Affects OpenApiConfig and SettingManager PostConstruct that retrieve settings during the startup,
+        // before the encryptor is initialized.
+        // TODO: Improve EncryptorInitializer. For now it depends on GeonetworkDataDirectory
+        //  that requires to be initialised in GeoNetwork start method.
+        if (!encryptor.isInitialized()) {
+            if (!se.get().isEncrypted()) {
+                value = se.get().getStoredValue();
+            } else {
+                throw new RuntimeException("Encrypted settings can't be accessed before encryptor is initialized");
```

Comment:
I'd rather use [IllegalStateException](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalStateException.html). The semantic is more specific for this case:
> Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation